shared_sql_utils/src/dsn.rs

```diff
@@ -118,26 +118,42 @@ impl DSNOpts {
         let buffer = &mut [0u16; MAX_VALUE_LENGTH];
         let mut dsn_opts = DSNOpts::default();
         let mut error_key = "";
-        self.iter().for_each(|(key, _)| {
-            let len = unsafe {
-                SQLGetPrivateProfileStringW(
-                    to_widechar_ptr(&self.dsn).0,
-                    to_widechar_ptr(key).0,
-                    to_widechar_ptr("").0,
-                    buffer.as_mut_ptr(),
-                    buffer.len() as i32,
-                    to_widechar_ptr(ODBCINI).0,
-                )
-            };
+        let len = unsafe {
+            SQLGetPrivateProfileStringW(
+                to_widechar_ptr(&self.dsn).0,
+                std::ptr::null(),
+                to_widechar_ptr("").0,
+                buffer.as_mut_ptr(),
+                buffer.len() as i32,
+                to_widechar_ptr(ODBCINI).0,
+            )
+        };
+        let dsn_keys = unsafe { input_text_to_string_w(buffer.as_mut_ptr(), len as usize) };
+        dsn_keys
+            .split('\0')
+            .filter(|s| !s.is_empty())
+            .collect::<Vec<&str>>()
+            .iter()
+            .for_each(|&key| {
+                let len = unsafe {
+                    SQLGetPrivateProfileStringW(
+                        to_widechar_ptr(&self.dsn).0,
+                        to_widechar_ptr(key).0,
+                        to_widechar_ptr("").0,
+                        buffer.as_mut_ptr(),
+                        buffer.len() as i32,
+                        to_widechar_ptr(ODBCINI).0,
+                    )
+                };
```

Comment:
I ended up going with scanning the registry for keys users had specified. This way we're not responsible for fetching `PWD`, `pWD`, and any other permutation. Instead, we fetch the keys now and trust `set_field` to handle the returned value correctly, or ignore it.